src/server/req_manager.c

```diff
@@ -3649,7 +3649,8 @@ struct batch_request *preq;
 
 	mymom   = pnode->nd_moms[0];
 	vn_pool = pnode->nd_attr[(int)ND_ATR_vnode_pool].at_val.at_long;
-	if (vn_pool > 0) {
+	if ((pnode->nd_attr[(int)ND_ATR_vnode_pool].at_flags & ATR_VFLAG_SET) && 
+	   (vn_pool > 0)) {
```

Comment:
Here's a suggestion.  Take it as you may.  I'd remove the vn_pool variable.  It looks a little strange to look at pnode on one side and vn_pool on the other.  The vn_pool variable is only used twice.  It'd also make this instance is like the others you changed.  Up to you.

Reply:
I see what you're saying.  However, since it is functionally the same, I will leave it as is.